sdk/src/test/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessorTest.java

```diff
@@ -213,17 +221,47 @@ public void serviceHandlerThrowsException() {
             .setScheduleDelayMillis(MAX_SCHEDULE_DELAY_MILLIS)
             .build());
     ReadableSpan span1 = createSampledEndedSpan(SPAN_NAME_1);
-    List<SpanData> exported = waitingSpanExporter.waitForExport(1);
+    List<SpanData> exported = waitingSpanExporter.waitForExport();
     assertThat(exported).containsExactly(span1.toSpanData());
-
+    waitingSpanExporter.reset();
     // Continue to export after the exception was received.
     ReadableSpan span2 = createSampledEndedSpan(SPAN_NAME_2);
-    exported = waitingSpanExporter.waitForExport(1);
+    exported = waitingSpanExporter.waitForExport();
     assertThat(exported).containsExactly(span2.toSpanData());
   }
 
+  @Test(timeout = 200)
+  public void exporterTimesOut() {
+    WaitingSpanExporter waitingSpanExporter =
+        new WaitingSpanExporter(1) {
+          @Override
+          public ResultCode export(List<SpanData> spans) {
+            ResultCode result = super.export(spans);
+            try {
+              // sleep longer than the configured timout of 100ms
+              Thread.sleep(1000);
+            } catch (InterruptedException e) {
+              // ok
```

Comment:
You may want to check that this happens otherwise we don't know if the test passed (currently we rely on the test timeout to do that.

 } catch (InterruptedException e) {
     return result;
}
fail if you get here.

PS: Happy to hear other opinion.

Reply:
that's a good idea. I'll see how it works out.

Reply:
it turned out to be slightly trickier than that, but you can see how I did it.